src/core/expr/fexpr_sumprod.cc

```diff
@@ -83,6 +83,7 @@ class FExpr_SumProd : public FExpr_Func {
         case SType::INT16:
         case SType::INT32:
         case SType::INT64:
+          col.cast_inplace(SType::INT64);
```

Comment:
You can do this, but then you need to also do the same for floats, etc, so better to revert the changes to what it was before this PR.

Reply:
This failure is not related to your PR:

```
    def test_dt_count_na2():
        DT = dt.Frame(G=[1,1,1,2,2,2], V=[None, None, None, None, 3, 5])
        EXP = dt.Frame(G=[1,2], V1=[3,1], V2=[1,0])
>       RES = DT[:, [dt.countna(f.V), dt.countna(dt.mean(f.V))], dt.by(f.G)]
E       AssertionError: Assertion 'i < nrows()' failed in src\core\column.cc, line 251
DT         = <Frame#229a5d3da80 6x2>

```

You need to disable this test until https://github.com/h2oai/datatable/issues/3417 is resolved.